src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -960,13 +973,49 @@ private void parseCommand() throws SnowflakeSQLException {
     // SNOW-15153: verify that the value after file:// is not changed by GS
     verifyLocalFilePath(localFilePathFromGS);
 
-    // more parameters common to upload/download
-    String stageLocation = jsonNode.path("data").path("stageInfo").path("location").asText();
-
     parallel = jsonNode.path("data").path("parallel").asInt();
 
     overwrite = jsonNode.path("data").path("overwrite").asBoolean(false);
 
+    stageInfo = getStageInfo(jsonNode);
+
+    if (logger.isDebugEnabled()) {
+      logger.debug("Command type: {}", commandType);
+
+      if (commandType == CommandType.UPLOAD) {
+        logger.debug("autoCompress: {}", autoCompress);
+
+        logger.debug("source compression: {}", sourceCompression);
+      } else {
+        logger.debug("local download location: {}", localLocation);
+      }
+
+      logger.debug("Source files:");
```

Comment:
Nit, this can probably be reduced to something like `logger.debug("Source files: {}", String.join(',', sourceFiles));`